File: ujian-ap2b.py
#Ajay Alfredo Almani_50420093_1IA16
print('============================')
print('UJIAN AP2B')
nama = input("Nama    : ")
npm  = input("NPM     : ")
kls  = input("Kelas   : ")
print('============================')
def menu():
    print('\n========PROGRAM MENU========')
    print('1. Biodata')
    print('2. Nilai')
    print('3. Program Kalkulator')
    print('4. Keluar')
    print('=============================')
    pil = int(input('Masukkan pilihan anda: '))
    print('=============================')

    if pil == 1:
        bio()
        menu()
    elif pil == 2:
        ratarata()
        menu()
    elif pil == 3:
        kalku()
        menu()
    elif pil == 4:
        exit()
    else:
        print('Maaf, Input yang Anda Masukkan salah! Silahkan Coba lagi.')
        menu()

def bio():
    nama = input("Masukkan Nama: ")
    npm  = input("Masukkan NPM: ")
    kls  = input("Masukkan Kelas: ")
    print('=====================================')


    print("Nama anda", nama)
    print("NPM anda", npm)
    print("Kelas anda", kls)
    menu()


def ratarata():
    uts  = int(input("Masukkan Nilai UTS : "))
    uas  = int(input("Masukkan Nilai UAS : "))
    
    jmlh = (uts * 0.4) + (uas * 0.6)
    print("\nNilai UTS Anda :", uts)
    print("Nilai UAS Anda :", uas)
    print("Total Nilai Anda :", jmlh)
    menu()

def kalku():
    a = int(input('Masukkan angka pertama : '))
    b = int(input('Masukkan angka kedua   : '))
    c = a + b
    d = a - b
    e = a * b
    f = a / b

    print(a, '+', b, '=', c)
    print(a, '-', b, '=', d)
    print(a, 'x', b, '=', e)
    print(a, ':', b, '=', f)

menu()

File: ujian-ap2b.rb
puts "--------------------------------"
puts "UJIAN AP2B"
print "Masukkan Nama : "
nama = gets.to_s
print "Masukkan Kelas : "
kelas = gets.to_s
print "Masukkan NPM : "
npm = gets.to_s
puts "--------------------------------"
def biodata()
  print "Masukkan Nama Anda = "
  nama=gets.chomp
  print "Masukkan NPM Anda = "
  npm=gets.to_i
  print "Masukkan Kelas Anda = "
  kls=gets.chomp
  puts"--------------------------------"
  puts"Saya #{nama}"
  puts"NPM Saya #{npm}"
  puts"Saya dari kelas #{kls}"
end

def nilai()
  print "Masukkan Nilai UTS = "
  uts = gets.to_i
  print "Masukkan Nilai UAS = "
  uas = gets.to_i
  
  totalnilai = (uts * 0.4) + ( uas * 0.6)
  puts"\nNilai UTS Saya #{uts}"
  puts"Nilai UAS Saya #{uas}"
  puts"Total Nilai Saya : #{totalnilai}"
end

 def kalkulator
 print "Masukkan Angka ke-1 ="
   a = gets.to_i
   print "Masukkan Angka ke-2 ="
   b= gets.to_i

   jmlh = a + b
   krng = a - b
   bagi = a.to_f / b.to_f
   kali = a * b
puts"#{a} + #{b} = #{jmlh}"
puts"#{a} - #{b} = #{krng}"
puts"#{a} * #{b} = #{kali}"
puts"#{a} / #{b} = #{bagi}"
   

 end
def menu()
puts '----------PROGRAM MENU----------'
puts '1.Biodata'
puts '2.Nilai'
puts '3.Kalkulator'
puts '4.Keluar'
puts '--------------------------------'
print 'Masukkan Pilihan = '
pil = gets.to_i
puts '--------------------------------'

if pil == 1
biodata()
menu()
elsif pil == 2
nilai()
menu()
elsif pil == 3
kalkulator()
menu()
elsif pil == 4 
exit()
else 
puts 'Maaf, Input yang Anda Masukkan salah! Silahkan Coba lagi.'
end
end

menu()